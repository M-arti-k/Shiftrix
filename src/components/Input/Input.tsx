import type { FC, JSX } from "react";
import type { IInput } from "src/interfaces/IInput";
import styles from "./Input.module.scss";

const Input: FC<IInput> = ({ type, children }): JSX.Element => {
  return (
    <div className={styles.underline}>
      <input
        data-testid="input"
        className={styles.input}
        type={type}
        placeholder={children}
      />
    </div>
  );
};

export default Input;
