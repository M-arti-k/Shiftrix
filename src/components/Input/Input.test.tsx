import { render, screen } from "@testing-library/react";
import Input from "./Input";

describe("Input test", () => {
  let input: HTMLInputElement;
  beforeEach((): void => {
    render(<Input type="email">Email</Input>);
    input = screen.getByTestId("input");
  });
  test("Input renders", (): void => {
    expect(input).toBeInTheDocument();
  });
  test("Input receives type from props", (): void => {
    expect(input).toHaveAttribute("type", "email");
  });
  test("Input receives children from props and displays in placeholder", (): void => {
    expect(input).toHaveAttribute("placeholder", "Email");
  });
  test("Input`s snapshot", (): void => {
    expect(input).toMatchSnapshot();
  });
});
