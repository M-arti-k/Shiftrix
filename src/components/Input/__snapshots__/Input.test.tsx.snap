// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Input\`s snapshot 1`] = `
<input
  class="input"
  data-testid="input"
  placeholder="Email"
  type="email"
/>
`;
